ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/ConvertTupleToArrayOrListElementsFix.fs

```diff
@@ -0,0 +1,43 @@
+﻿namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open System.Text.RegularExpressions
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Impl
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Parsing
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Psi.ExtensionsAPI.Tree
+open JetBrains.ReSharper.Psi.Util
+open JetBrains.ReSharper.Resources.Shell
+
+type ConvertTupleToArrayOrListElementsFix(expr: IFSharpExpression, actualType: string) =
+    inherit FSharpQuickFixBase()
+
+    let regex = Regex("'\w\d*( \* '\w\d*)+")
+    let arrayOrListExpr = ArrayOrListExprNavigator.GetByExpression(expr.IgnoreParentParens())
+    let isArray = arrayOrListExpr :? IArrayExpr
+
+    new (warning: AddTypeEquationError) =
+        ConvertTupleToArrayOrListElementsFix(warning.Expr, warning.ActualType)
+
+    override x.Text =
+        if isArray then "Convert tuple to array elements" else "Convert tuple to list elements"
```

Comment:
Just an idea: the reported error is about using a wrong type, and we guess it happens due to a wrong separator use. For a newcomer it may be not that obvious why the existing code is not a few list/array elements (in fact, the tuple _is_ such an element!) or why do we want to convert it. What if the text was something like `"Use ';' separators"`? Would it be easier to grasp what went wrong and why it would fix the code?

Reply:
good idea, thanks!

Reply:
But we also have the non-standard case ``[(a, b, c)]`` - it is not enough just to replace the separators with ';' but also get rid of the explicit parentheses

Reply:
Good point! I'm OK with either ignoring such cases or removing the parens.